packages/vite/src/node/plugins/define.ts

```diff
@@ -53,12 +53,11 @@ export function definePlugin(config: ResolvedConfig): Plugin {
     }
 
     const replacements: Record<string, string> = {
-      ...processNodeEnv,
+      ...(!ssr || config.ssr?.target === 'webworker' ? processNodeEnv : {}),
```

Comment:
I think we should refactor the `!ssr || config.ssr?.target === 'webworker'` part into a variable as it's used on line 47 as well.

Reply:
good point!